assignments/lambda-classes.js

```diff
@@ -1 +1,205 @@
-// CODE here for your Lambda Classes
+/**
+ * A class to represent a person at school.
+ * @class
+ * @constructor
+ * @property name the person's name
+ * @property age the person's age
+ * @property location the person's location
+ * @property gender the person's gender
+ * @function speak() return a string  
+ */
+
+class Person {
+    constructor(obj) {
+        this.name = obj.name;
+        this.age = obj.age;
+        this.location = obj.location;
+        this.gender = obj.gender;
+    }
+    speak() {
+        return `Hello, my name is ${this.name}, I am from ${this.location}`;
+    }
+} 
+
+/**
+ * A class to represent an instructor at school.
+ * @class
+ * @constructor
+ * @augments Person
+ * @property specialty the instructor's specialty
+ * @property favLanguage the instructor's favLanguage
+ * @property catchPhrase the instructor's catchPhrase
+ * @function demo() 
+ *      @param subject 
+ *      @returns string '​​​​​Today we are learning about {subject}' 
+ * @function grade() 
+ *      @param studentObj
+ *      @param subject
+ *      @returns string '​​​​​{student.name} receives a perfect score on {subject}'
+ * @function editStudentGrade() utility function, modify student object's grade
+ *      @param studentObj
+ *      @returns string `{Instructor/PM name} just changed {student name}'s grade from {previous grade} to {new grade}`
+ */
+class Instructors extends Person {
+    constructor(instructorObj){
+        super(instructorObj);
+        this.specialty = instructorObj.specialty;
+        this.favLanguage = instructorObj.favLanguage;
+        this.catchPhrase = instructorObj.catchPhrase;
+    }
+    demo(subject) {
+        return `Today we are learning about ${subject}`;
+    }
+    grade(studentObj, subject) {
+        return `${studentObj.name} receives a perfect score on ${subject}`;
+    }
+    static editStudentGrade(studentObj) {
```

Comment:
Relies upon a object parameter with the key name. I'm interested in seeing how this is used staticly, I did this sometimes in order to skip instancing.